tock/employees/models.py

```diff
@@ -38,6 +40,23 @@ class EmployeeGrade(models.Model):
     def __str__(self):
         return '{0} - {1} (Starting: {2})'.format(self.employee, self.grade, self.g_start_date)
 
+    def get_grade(end_date, user):
+        """Gets grade information based on a date and user. Original use of
+        function is to append correct grade information to each
+        TimecardObject."""
+        emp_grd_objects = EmployeeGrade.objects.filter(
+                Q(employee = user)
+                & Q(g_start_date__lte = end_date)
+                ).all().aggregate(Max('g_start_date'))['g_start_date__max']
+
+        if emp_grd_objects:
```

Comment:
So this is kind of an oddball way to be going about this--we're checking if the user has grade objects, then, if they do, separately querying again to pull the one we want. It'd be more straightforward to just use get() to fetch the EmployeeGrade object we want, and set the grade to None if the object does not exist.